tools/mapmerge2/mapmerge.py

```diff
@@ -86,7 +87,27 @@ def select_key(assigned):
 def main(settings):
     for fname in frontend.process(settings, "merge", backup=True):
         shutil.copyfile(fname, fname + ".before")
-        old_map = DMM.from_file(fname + ".backup")
+        try:
+            old_map = DMM.from_file(fname + ".backup")
+        except FileNotFoundError:
+            print("\nError: Could not find a backup file!\n")
+            print("If you have already committed your changes, enter the git hash PREVIOUS to the commit.\n" \
+                  + "    Otherwise, enter \"HEAD\".\n" \
+                  + "    Enter \"(q)uit\" to exit.")
+            githash = input(">> ")
+            if (githash.lower() == "q") or (githash.lower() == "quit"):
+                return
+            
+            try:
+                # git show 123abc^:path/to/map.dmm > path/to/map.dmm.backup
+                gitfilename = fname.replace("\\","/") # Convert from windows backslash if needed
+                p = subprocess.Popen("git show " + githash + "^:" + gitfilename + " > " + (fname + ".backup"), stdout=subprocess.PIPE, shell=True)
+                p.wait()
+                old_map = DMM.from_file(fname + ".backup")
+            except OSError: # command git not found
+                print("An error occurred. Is Git installed?")
```

Comment:
A few people do download the zip directly. They hopefully learn to use git sometimes they actually just download it to host and play with friends directly and not much else.